### ISSUE TITLE
using extract on trusted data will show undefined

### ISSUE THREAD

**Issue by ghost** [sirbrillig/phpcs-variable-analysis#240]
Is there a way to ignore/identify variables coming from php's method `extract()`. We utilize this in many places for readability and simplicity. Recently we started using `PHP_Sniffer` with variable analysis, all of our extracted `vars` show as undefined. If you could point me in the right direction, that would be very nice.

**Comment by sirbrillig** [comment]
`extract()` is a difficult function to analyse with a system like phpcs because it dynamically creates variables at runtime; it's hard to know by just looking at the code what it will do. For example:

```php
function doSomething( $data ) {
  extract( $data );
  return $name;
}
```

In the example above, should we mark `$name` as undefined or not? We can't know.

Full static analysis systems like `phpstan` or `psalm` may be able to handle these situations because they can follow the full flow of your code to determine every possible outcome, but this sniff is designed to operate on isolated chunks of code or single files; it doesn't know what your data actually is. If you're committed to using dynamic variable assignment, you might want to explore static analysis tools as an option instead.

I'm going to close this but feel free to reopen if you have additional questions.

**Comment by ghost** [comment]
Wow, you really know your stuff! Thank you @sirbrillig for the quick turn around.
